libs/designsystem/src/lib/components/radio/radio-group/radio-group.component.ts

```diff
@@ -236,10 +236,15 @@ export class RadioGroupComponent implements AfterContentInit, ControlValueAccess
   }
 
   private refreshSelectionState() {
-    if (this.value) {
+    if (this.value || this.value === false) {
       this._selectedIndex = this.getIndexOfSelectedValue(); // Ensure selectedIndex reflects value within items
     }
-    this._value = this.getValueFromSelectedIndex() || null; // Ensure value exists within items
+
+    const valueFromSelectedIndex = this.getValueFromSelectedIndex();
+    this._value =
+      !valueFromSelectedIndex && valueFromSelectedIndex !== false
+        ? valueFromSelectedIndex || null
+        : valueFromSelectedIndex; // Ensure value exists within items
```

Comment:
I find this logically weird to look at :)

In a simplified form it reads:

_Value is equal to either A or null or A_

The expression `!valueFromSelectedIndex && valueFromSelectedIndex !== false` is also a bit odd - probably due to the fact that we're trying to test if `valueFromSelectedIndex` is falsy but not false.

The same goes for the remaining two changes in this commit.

I suggest we huddle up and put "hovederne i blød" - maybe dust off the old boolean algebra and Demorgan's laws 😁

Reply:
Lol had to spend ~30 seconds too long to mentally parse my own code again. I will take a look at it 😅

Reply:
I must have had a stroke or something when I wrote it. Here's something that hopefully communicates intent much clearer! 

``` Typescript   
this._value =
      valueFromSelectedIndex || typeof valueFromSelectedIndex === 'boolean'
        ? valueFromSelectedIndex
        : null; // Ensure value exists within items
```

Reply:
I like it, maybe with a comment for later detailing why these checks are necessary, so we dont have to guess the reasoning next time, like you had to this time around.

Reply:
I've refactored the bug fix. I think it's a lot cleaner now and intuition shines through more clearly.